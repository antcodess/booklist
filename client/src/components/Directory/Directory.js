import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import DirectoryItems from '../DirectoryItems/DirectoryItems';

import { selectShopBestSellers, 
         selectShopNewReleases, 
         selectShopComingSoon } from '../../redux/shop/shop.selectors';

import './Directory.scss';

export const Directory = ({ bestSellers, newReleases, comingSoon }) => {
  return (
    <section className='directory'>
      <DirectoryItems title='Best Sellers' items={bestSellers} />
      <DirectoryItems title='New Releases' items={newReleases} />
      <DirectoryItems title='Coming Soon' items={comingSoon} />
    </section>
  );
};

const mapStateToProps = createStructuredSelector({
  bestSellers: selectShopBestSellers,
  newReleases: selectShopNewReleases,
  comingSoon: selectShopComingSoon
});


export default connect(mapStateToProps)(Directory);