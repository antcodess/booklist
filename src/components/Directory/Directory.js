import React from 'react';
import DirectoryItems from '../DirectoryItems/DirectoryItems';

import './Directory.scss';

const Directory = () => {
  return (
    <section className='directory'>
      <DirectoryItems title='Best Sellers' />
      <DirectoryItems title='New Releases'/>
      <DirectoryItems title='Coming Soon' />
    </section>
  );
};

export default Directory;