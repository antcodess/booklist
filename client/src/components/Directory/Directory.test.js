import React from 'react';
import { shallow } from 'enzyme';
import { Directory } from './Directory';

describe('Directory Component', () => {
  let wrapper;
  let mockBestSellers;
  let mockNewReleases;
  let mockComingSoon;
  let mockCategories;

  beforeEach(() => {

    mockBestSellers = [{id: 1}, {id: 2}, {id: 3}];
    mockNewReleases = [{id: 1}, {id: 2}, {id: 3}];
    mockComingSoon = [{id: 1}, {id: 2}, {id: 3}];

    mockCategories = [mockBestSellers, mockNewReleases, mockComingSoon]

    const mockProps = {
      bestSellers: mockBestSellers, 
      newReleases: mockNewReleases, 
      comingSoon: mockComingSoon
    };

    wrapper = shallow(<Directory { ...mockProps } />);
  });

  it('should render Directory Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correct amount of DirectoryItems as categories length', () => {
    expect(wrapper.find('.directory').children().length).toEqual(mockCategories.length);
  });
});