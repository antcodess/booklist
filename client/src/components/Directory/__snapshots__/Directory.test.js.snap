// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory Component should render Directory Component 1`] = `ShallowWrapper {}`;
